pkgs/development/tools/scalafmt/default.nix

```diff
@@ -1,29 +1,34 @@
-{ stdenv, fetchurl, unzip, jre }:
+{ stdenv, jre, fetchMavenArtifact, makeWrapper, gnugrep }:
 
+let
+  support = (import ./support.nix) fetchMavenArtifact;
+  classpath = stdenv.lib.concatStringsSep ":" (map (dep: dep.jar) support.deps);
```

Comment:
collecting of jars from dependencies is done by [set-java-classpath.sh](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/set-java-classpath.sh), so this line could be omitted, like this https://github.com/volth/nixpkgs/commit/1e4a28a726bb1ea83bd78755cb31e7be6a0a1058

Reply:
Ah good to know, but shouldn't it actually also run when only `jre` is in `buildInputs` ? Because now we needlessly depend on the whole `jdk` instead of the smaller `jre`

Reply:
> Because now we needlessly depend on the whole jdk

Only for the build, there should be no runtime dependency